### ISSUE TITLE
Fix fibo_rec.vat

### ISSUE THREAD

**Issue by jarctan** [jarctan/vache#18]
```  
error[E0597]: `i` does not live long enough
      --> src/main.rs:799:47
       |
   771 | pub fn fibo<'a, 'b>(
       |                 -- lifetime `'b` defined here
   772 |     mut i: Var<'a, 'b, __Integer>,
       |     ----- binding `i` declared here
   773 | ) -> __Result<__Ret<Cow<'b, __Integer>, __fiboRet>> {
   774 |     let __res: __Result<Cow<'b, __Integer>> = try {
       |                ---------------------------- type annotation requires that `i` is borrowed for `'b`
   ...
   799 |                                         __ref(&i),
       |                                               ^^ borrowed value does not live long enough
   ...
   824 | }
       | - `i` dropped here while still borrowed
   
   error[E0597]: `i` does not live long enough
      --> src/main.rs:809:47
       |
   771 | pub fn fibo<'a, 'b>(
       |                 -- lifetime `'b` defined here
   772 |     mut i: Var<'a, 'b, __Integer>,
       |     ----- binding `i` declared here
   773 | ) -> __Result<__Ret<Cow<'b, __Integer>, __fiboRet>> {
   774 |     let __res: __Result<Cow<'b, __Integer>> = try {
       |                ---------------------------- type annotation requires that `i` is borrowed for `'b`
   ...
   809 |                                         __ref(&i),
       |                                               ^^ borrowed value does not live long enough
   ...
   824 | }
       |  - `i` dropped here while still borrowed
```

```rust
fn fibo(i: int) -> int {
    if i == 0 {
        0
    } else if i == 1 {
        1
    } else {
        fibo(i - 1) + fibo(i - 2)
    }
}

fn main() {
    print(fibo(0));
    print(fibo(1));
    print(fibo(2));
    print(fibo(11));
}
```

I think it's not working because of the definition of the lifetimes in 
```rust
pub(crate) fn __ref<'d, 'e, 'a: 'd + 'e, 'b: 'd + 'e, 'c: 'd + 'e, B: Clone>(
    var: &'c Var<'a, 'b, B>,
) -> Var<'d, 'e, B> {
/* */
}
```

bc clearly there is nothing wrong with returning a reference to i in the function `fibo` since `i` is an argument of the fn

**Comment by jarctan** [comment]
Actually changing the signature of the function to:
```
pub fn fibo<'a, 'b, 'd>(
    mut i: Var<'a, 'b, __Integer>,
) -> __Result<__Ret<Cow<'d, __Integer>, __fiboRet>> {
    let __res: __Result<Cow<'d, __Integer>> = try {
```
fixes the issue.
The thing is probably that the result does not have the same lifetime as the argument.

**Comment by jarctan** [comment]
Adding a new lifetime parameter `'d` is in contradiction with test exec::check_modes_va__min_max

If `'a` and `'b` are the lifetime of the arguments, and `'c` the lifetime of the return typye, then: 
* exec::check_modes_va__min_max requires `'b: 'c`
* cargo test exec::exec_va__fibo_rec requires `'c: 'b`, `'c: 'a` or `'c` alone

Thus, we cannot have a bound for the return type that fits all

**Comment by jarctan** [comment]
Fixed by sledgehammer commit 2fd11bdc716c457b3175026179577d265d3e37e7
